Source/ASTextNode2.mm

```diff
@@ -350,6 +354,8 @@ - (void)prepareAttributedString:(NSMutableAttributedString *)attributedString
     shadow.shadowBlurRadius = _shadowRadius;
     [attributedString addAttribute:NSShadowAttributeName value:shadow range:NSMakeRange(0, attributedString.length)];
   }
+  _preparedAttributedText = [attributedString copy];
```

Comment:
This second copy is not necessary, right? Since this code is called so much, I'd suggest we just let the mutable object live on, as anyone up-casting it can crash as much as they want :).

Reply:
So my thing in cases like this, is if I think there's even one downstream copy that is likely, just go ahead and copy now to make that free, to avoid multiple copies of the same object, and to save my mental energy. This method will get called once per node per configuration (change of properties with a layout/render pass in between). So in one data controller batch, this should be called realistically <100 times (say, 25 items with 4 text nodes per).

In this case for example, the attributed string will get copied when we make an ASTextNodeCacheKey from it anyway. And in fact, it will get copied each layout/render pass, so we will do more copying if we don't change properties in between those passes. We could remove THAT copy too, but then we're putting a little crack in our abstractions. These copies will be fast, they will mostly happen off-main, and there will only be ~1 per text node in common cases so I think we should leave it.